examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -0,0 +1,173 @@
+#include "static-supported-modes-manager.h"
+#include "ESP32Config.h"
+#include <app/util/debug-printing.h>
+#include <app/util/ember-print.h>
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer::Internal;
+using namespace chip::app::Clusters::ModeSelect;
+using chip::Protocols::InteractionModel::Status;
+
+using ModeOptionStructType = Structs::ModeOptionStruct::Type;
+using SemanticTag          = Structs::SemanticTagStruct::Type;
+template <typename T>
+using List                                                                                = app::DataModel::List<T>;
+
+StaticSupportedModesManager::ModeLabel* StaticSupportedModesManager::modeLabelList = nullptr;
+StaticSupportedModesManager::ModeOptionStructType* StaticSupportedModesManager::modeOptionStruct = nullptr;
+StaticSupportedModesManager::SemanticTag* StaticSupportedModesManager::semanticTags = nullptr;
+
+const StaticSupportedModesManager StaticSupportedModesManager::instance = StaticSupportedModesManager();
+
+void *StaticSupportedModesManager::endpointArray[3][2];
+
+void StaticSupportedModesManager::InitEndpointArray()
+{
+	for(int i=0; i<3; i++) {
+		endpointArray[i][0] = nullptr;
+		endpointArray[i][1] = nullptr;
+	} 
+}
+
+SupportedModesManager::ModeOptionsProvider StaticSupportedModesManager::getModeOptionsProvider(EndpointId endpointId) const
+{
+	if(endpointArray[endpointId][0] != nullptr && endpointArray[endpointId][1] != nullptr) {
```

Comment:
OK, so the "3" is "number of endpoints".  And this is assuming all consumers will have endpoint _ids_ (which can be arbitrary!) in the range 0-2.  Why is that an OK assumption?  And why is failing catastrophically (read: critical security bug due to writing to random memory) when that assumption is violated ok?

Reply:
Used FIXED_ENDPOINT_COUNT for the array